wandb/sdk/artifacts/artifact_manifest_entry.py

```diff
@@ -150,7 +150,11 @@ def download(
 
         # Skip checking the cache (and possibly downloading) if the file already exists
         # and has the digest we're expecting.
-        if os.path.exists(dest_path) and self.digest == md5_file_b64(dest_path):
+        if (
+            os.path.exists(dest_path)
+            and not os.path.isdir(dest_path)
+            and self.digest == md5_file_b64(dest_path)
+        ):
```

Comment:
suggestion: use `try/except` instead of checking `path.exists/path.isdir` -- this guards against race conditions, at least within this code.

Looks like `md5_file_b64` ends up raising the "expected" errors (which is convenient!), i.e.
- `FileNotFoundError` if dest_path doesn't exist
- `IsADirectoryError` if dest_path is a directory

```suggestion
        try:
            md5_hash = md5_file_b64(dest_path)
        except (FileNotFoundError, IsADirectoryError):
            pass
        else:
            if self.digest == md5_hash:
                return FilePathStr(dest_path)
```